File: FizzBuzz/erlang/src/FiltersProvider.erl
-module('FiltersProvider').

-export([addFilter/1, numberOfFilters/0, removeAllFilters/0, filters/0]).

addFilter(Filter) ->
    'ServiceProvider':add(Filter, ?MODULE).

numberOfFilters() ->
    'ServiceProvider':count(?MODULE).

removeAllFilters() ->
    'ServiceProvider':removeAll(?MODULE).

filters() ->
    'ServiceProvider':services(?MODULE).

File: FizzBuzz/JavaScript/src/FiltersProvider.js
const filters = new Set();

class FiltersProvider {
    add(filter) {
	addFilter(filter);
	return filters.size;
    }

    removeAll() {
	 filters.clear();
	 return filters.size;
    }
}

const addFilter = (filter) => {
    filters.add(filter);
};

module.exports = FiltersProvider;
